timelock-agent/src/main/java/com/palantir/timelock/paxos/DbBoundTimestampCreator.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.timelock.paxos;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import com.google.common.base.Preconditions;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.factory.ServiceDiscoveringAtlasSupplier;
+import com.palantir.atlasdb.spi.KeyValueServiceConfig;
+import com.palantir.atlasdb.timelock.paxos.DelegatingManagedTimestampService;
+import com.palantir.atlasdb.timelock.paxos.ManagedTimestampService;
+import com.palantir.timestamp.TimestampManagementService;
+import com.palantir.timestamp.TimestampService;
+
+public class DbBoundTimestampCreator implements TimestampCreator {
+
+    private KeyValueServiceConfig kvsConfig;
+
+    public DbBoundTimestampCreator(KeyValueServiceConfig kvsConfig) {
+        this.kvsConfig = kvsConfig;
+    }
+
+    @Override
+    public Supplier<ManagedTimestampService> createTimestampService(String client) {
+        ServiceDiscoveringAtlasSupplier atlasFactory = new ServiceDiscoveringAtlasSupplier(kvsConfig,
+                Optional.empty(),
```

Comment:
We should check with @datval but my guess is that it's fine to simply not support using Cassandra with the DbBound timestamp stuff. Actually, I think I'd prefer that if we could to prevent Foundry folks from using this.

Reply:
I talked to the internal large product team and indeed the timestamp table will not live on C* in any case. I have added the leader block here, and will add docs stating that we do not recommend using this, unless you want to persist all state in the DB. Should we really drop C*  support here, creating more snowflakiness?